src/hatch/publish/auth.py

```diff
@@ -72,6 +73,12 @@ def __get_username(self) -> str:
         self.__username_was_read = True
         return self._app.prompt(f"Username for '{self._repo_config['url']}' [__token__]") or '__token__'
 
+    def _read_keyring(self) -> str | None:
```

Comment:
this could also store the `__password` property as i'm assuming that using a username from this credentials source doesn't make sense with a password from another source.

Reply:
You are right, no need to run read the password again if we already have it. I pushed my changes. Thanks for the review